.ci/azure-pipelines/tidying.yaml

```diff
@@ -0,0 +1,29 @@
+jobs:
```

Comment:
@shrijitsingh99 Could you please check this? You've played with GitHub actions more than me. This feels like Azure format to me.

Also, @gnawme if you could run this on your fork and link the results that'd be great

Reply:
Yeah, this is an azure format plus github actions are meant to be in `.github/workflow`

Reply:
Probably moot, since @kunaltyagi stated a preference for integrating `clang-tidy` into CMake versions that support it directly?

But yes, I think I copied an Azure script. I'll revisit this.

Reply:
> Probably moot, ... integrating clang-tidy into CMake versions that support it directly

We can switch to a pure CMake version, but adding the GitHub Action for compile-commands + bash should not be hard. We can make an issue to track the cmake integration as a separate actionable item (which based on your previous comment, it is)